File: cppVSjava/Parameterized_inheritance/parin.java
class stud
{
    public int sr_no;
    public stud(int no)
    {
        System.out.println("Super class constructor called...");
        sr_no=no;
    }
}
class marks extends stud
{
    public int er_no;
    public marks(int a,int b)
    {
        super(b);                                     //*****_________******//
        System.out.println("Subclass constructor called...");
        er_no=a;
    }
}
public class Parajav {
    public  static void main(String args[])
    {
        marks ob1=new marks(10,20);
    }
}

File: cppVSjava/Parameterized_inheritance/parin.cpp
#include <iostream.h>
#include <conio.h>
class stud
{
private:
	int s_no;
public:
	stud()
	{
		cout<<"Default constructor called...\n";
	}
	stud(int s_no)
	{
		cout<<"Stud class constructor...\n";
		this->s_no=s_no;
	}
} ;
class marks:public stud
{
private:
	int er_no;
public:
	marks(int er_no,int no)
	{
		stud::stud(no);					//*****________*****//
		this->er_no=er_no;
		cout<<"Base class constructor called..\n";
	}
} ;

void main()
{
	marks ob1(10,20);
	getch();
}
